src/pages/presentations.js

```diff
@@ -8,79 +8,44 @@ import Hero from '../components/hero'
 import Container from '../components/container'
 import SearchBar from '../components/searchbar'
 import LectureCard from '../components/lecturecard'
-import { useCallback, useState } from 'react'
-
-const debounce = (func, wait) => {
-  let timeout
-
-  return (...args) => {
-    const later = () => {
-      clearTimeout(timeout)
-      func(...args)
-    }
-    clearTimeout(timeout)
-    timeout = setTimeout(later, wait)
-  }
-}
-
-const CardGrid = ({ lectures }) => {
-  return (
-    <Grid
-      sx={{
-        gridTemplateColumns: [
-          'repeat(auto-fill, minmax(200px, 1fr))',  // better way to do this?
-          null,
-          'repeat(auto-fill, minmax(300px, 1fr))',
-        ],
-      }}
-    >
-      {lectures.map(({ node: lecture }, i) => (
-        <LectureCard
-          key={i}
-          title={lecture.title}
-          body={lecture.body}
-          level={lecture.level}
-          link={lecture.link}
-          date={lecture.date}
-        />
-      ))}
-
-    </Grid>
-  )
-}
+import { useRef, useState } from 'react'
+import debounce from '../utils/debounce'
+import CardGrid from '../components/cardgrid';
 
 const Presentations = ({ data }) => {
   const {
     allLecturesYaml: {
-      edges: lectures,
+      nodes: lectures,
     },
     allLectureFoldersYaml: {
-      edges: lectureFolders,
+      nodes: lectureFolders,
     },
   } = data
 
-  const { node: currentFolder } = lectureFolders[0]
+  const currentFolder = lectureFolders[0]
 
   const [pattern, setPattern] = useState('')
   const [displayedLectures, setDisplayedLectures] = useState(lectures)
 
   const fuseOptions = {
-    keys: [{name: 'node.title', weight: 2}, 'node.body'],
+    keys: [{name: 'title', weight: 2}, 'body'],
     threshold: 0.4,
   }
-  const fuse = new Fuse(lectures, fuseOptions)
+  const fuse = useRef(new Fuse(lectures, fuseOptions)).current
 
-  const search = useCallback(debounce((value) => {
+  const search = useRef(debounce((value) => {
```

Comment:
We can also maybe use something like lodash for our debouncing, and convert it into a custom hook if someone hasn't already

Reply:
yeah lodash is fine, though I copied this debounce from some blog post so it's 100% reliable